Framework/Nexus/src/MuonNexusReader.cpp

```diff
@@ -122,7 +122,48 @@ void MuonNexusReader::readFromFile(const string &filename) {
   // get instrument name
   handle.openGroup("instrument", "NXinstrument");
   handle.readData("name", nexus_instrument_name);
-  handle.closeGroup();
+
+  // Try to read in period information
+  std::vector<std::string> periodInformation{
+      "period_sequences",  "period_labels",       "period_type",  "frames_period_requested",
+      "frames_period_raw", "total_counts_period", "period_output"};
+  try {
+    handle.openGroup("beam", "NXbeam");
+    for (const auto &info : periodInformation) {
```

Comment:
Might this be cleaner to split out the log names and group them by type in separate arrays, then we wouldn't have to have the `if` checks and duplicate the names in the loop and we'd just run separate loops for each?